### DISCUSSION TITLE
Updating CSS

### DISCUSSION THREAD

**Discussion by benlunt** [spatie/laravel-dashboard#71] [Q&A] [answered]
Hello,  Firstly can I say you've done a great job with the dashboard. Easy to install, understand and create new tiles. This isn't really an issue but some help please as the instructions are little unclear.

I am however currently struggling with how I would update the css that is within the Spatie vendor package. I am wanting to add another font size to the css.
eg.
fontSize: {
			**'xxs': '.65rem',**
	       'xs': '.75rem',
	       'sm': '.875rem',
	       'tiny': '.875rem',
			'base': '1rem',
			'lg': '1.125rem',

 I thought i could add it to the Spatie tailwind.config.js as i'd like it across the whole of the dashboard. I then can't work out how to update the dashboard.min.css file which is produced.  Apolgoie if this is an easy answer but i would appreicate some direction. 

I'm also unsure as to how i would add extra css using Dashboard::stylesheet($urlToStyleSheet); where do I put this line as in the documentation it says Spatie\Dashboard\Facades\Dashboard but i have tried that and nothing is happening? 

Really appreciate your help.

**Comment by willemvb** [comment, marked as answer]
Hi @benlunt,

Make sure you have everything installed by running `npm install` or `yarn install`, depending on which package manager you prefer.
You can then run `yarn run build` or `npm run build` to generate a new `dashboard.min.css`.

As for your second question, I'd like to ping @freekmurze for some backend insight!

**Reply by freekmurze** [reply]
> I'm also unsure as to how i would add extra css using Dashboard::stylesheet($urlToStyleSheet); where do I put this line as in the documentation it says Spatie\Dashboard\Facades\Dashboard but i have tried that and nothing is happening?

Put that in a service provider. The `AppServiceProvider` might be a good candidate.

**Reply by benlunt** [reply]
Thats for your super quick reply I wasn't expecting that. 

I am using NPM. I have installed Laravel and then brought in your package. Everything seems to work fine in creating a custom tile and updating the theme to light an dark. Its simply this bit which brings the spatie\dist\dashboard.min.css into an inline style variable assets.  

dashboard.blade.php

 <title>Dashboard</title>
        <meta name="google" value="notranslate">
        <meta name="csrf-token" content="{{ csrf_token() }}">
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">

        **{{ $assets }}**

        @stack('assets')

        <livewire:styles />

I ran the npm run build but its showing as npm ERR! missing script: build. Is this something i have update? Apologies but i don't understand how I can rebuild the css part. Should the npm build command do this?

**Reply by willemvb** [reply]
Which version of npm are you using? Latest version `7.8.0` should be working.
Have you run `npm install` in the project? This should add Tailwind and postCSS amongst others to your project.

After that, `npm run build` should perform this command: https://github.com/spatie/laravel-dashboard/blob/master/package.json#L5 to rebuild the production CSS files

**Reply by benlunt** [reply]
I have updated my version of npm to 7.8.0. I have ran the npm install and npm update within the project just to make sure. I am using composer 2.0.12. Still no joy with the tailwind and postcss file being populated within my project file. Is there something i am missing. 

The npm run build still is missing. Can i check do i run all of this in the main laravel project and not in the vendor package?

Thanks

**Reply by sebastiandedeyne** [reply]
Unfortenately there isn't a way to _replace_ assets at the moment, you can only add. Your best bet is to publish the main `dashboard.blade.php` view and override what's in `<head>` and drop `{{ $assets }}`, but that might give problems with other tiles…

This is an area we should improve on. We probably won't be working on this ourselves, but I created an [issue](https://github.com/spatie/laravel-dashboard/issues/72) outlining the current problems so maybe someone else can pick it up in the near future :)

**Reply by npalmqvist** [reply]
> I have updated my version of npm to 7.8.0. I have ran the npm install and npm update within the project just to make sure. I am using composer 2.0.12. Still no joy with the tailwind and postcss file being populated within my project file. Is there something i am missing.
> 
> The npm run build still is missing. Can i check do i run all of this in the main laravel project and not in the vendor package?
> 
> Thanks

Having the same issue. 
Just getting `npm ERR! missing script: build` when trying the above command. 
Just want to recompile tailwind assets since some classes seem to be missing in the precompiled assets.

**Reply by sebastiandedeyne** [reply]
Recompiling the included Tailwind assets is currently not supported. You'll need to compile your own CSS file, and override `dashboard.blade.php` as explained above.
